models/company_gateway/models/llm/llm.py

```
@@ -0,0 +1,663 @@
+                raise CredentialsValidateFailedError(
+                    f"Validation failed with status {response.status_code}: {self._extract_error_message(response)}"
+                )
+                
+            response_data = response.json()
+            if 'choices' not in response_data:
+                raise CredentialsValidateFailedError("Invalid response format from gateway")
+                
+        except Exception as ex:
+            if isinstance(ex, CredentialsValidateFailedError):
+                raise ex
+            raise CredentialsValidateFailedError(str(ex))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)